### DISCUSSION TITLE
How to handle document ID creation for cloud firestore?

### DISCUSSION THREAD

**Discussion by vegetablesalad** [ToolJet/ToolJet#9295] [Q&A]
When adding document to cloud firestore it needs document ID. For example adding doc to "/users/", the query builder will fail if I don't set some kind of "/users/myid". How can I enable auto ID generation, or should I do it with JS, but then I would need to check if the ID is unique.

**Comment by vegetablesalad** [comment]
So far I'm using :
`/users/{{Array.from(Array(20), () => Math.floor(Math.random() * 36).toString(36)).join('')}}`
But I'm sure this is not the intended way to solve this.

**Comment by ShazanRizvi** [comment]
Hello @vegetablesalad 
For implementation-related questions, our Slack community is recommended as our engineers are active there. Here is the invite: https://tooljet.com/slack.
